src/c/profile/session/read_access.c

```diff
@@ -98,10 +98,7 @@ void read_submessage_format(uxrSession* session, ucdrBuffer* data, uint16_t leng
 inline void read_format_data(uxrSession* session, ucdrBuffer* ub, uint16_t length,
                       uxrStreamId stream_id, uxrObjectId object_id, uint16_t request_id)
 {
-    uint32_t offset;
-    ucdr_deserialize_uint32_t(ub, &offset); //Remove this when data serialization will be according to the XRCE spec.
-    length = (uint16_t)(length - 4); //by the offset. Remove too with the future serialization according to XRCE
-
+    (void) length;
```

Comment:
Why do you not remove the length argument?

Reply:
This length is no computed from the data size removed. It contains a valid length. Before the fragmentation, it was used to limit the ucdrBuffer, but now it can not be used because of the fragmentation and the linked buffer feature of MicroCDR. However, it is still there because the value is valid and in the future it can be used to check something or for security purposes.

Reply:
Ok, I prefer to remove unnecessary arguments even though I need them in the future. But it is just a matter of taste.